### ISSUE TITLE
Files missing from rikaichamp-x.y.z-src.zip source package

### ISSUE THREAD

**Issue by cameel** [birtles/rikaichamp#600]
This is somewhat related to #599.

I see three source packages listed on the release pages. For example for 0.5.5:
- `rikaichamp-0.5.5-src.zip`
- `Source code (zip)`
- `Source code (tar.gz)`

The last two are generated automatically by github while the first one is custom-built so my assumption would be that it's added there because there's something wrong with the automatically generated ones. When trying to build the package only the automatically generates ones work though. `rikaichamp-0.5.5-src.zip` is missing `tsconfig.json` which results in multiple errors during `yarn build`:

```
  ERROR in ./src/background.ts 
  [tsl] ERROR
        TS18002: The 'files' list in config file 'tsconfig.json' is empty.

  ERROR in ./src/background.ts
  Module build failed (from ./node_modules/ts-loader/index.js):
  Error: error while parsing tsconfig.json
      at Object.loader (/tmp/firefox-extension-rikaichamp/src/node_modules/ts-loader/dist/index.js:17:18)
```

The archive is also missing `manifest.json`, though there is `manifest.json.src` which looks like a template to generate it. It suggests that maybe I just need to run one of the `yarn` scripts to get the missing files but I don't see anything that would be a template for `tsconfig.json` and AFAIK it's not something that's generated automatically.

**Comment by birtles** [comment]
Yes, you're right. It should include `tsconfig.json`. The manually generated source package is simply there because submitting to addons.mozilla.org requires supplying a source package for code review (as opposed to actually using it to build anything). Perhaps the automatically generated source packages are sufficient, however.

As for `manifest.json`, it is generated when webpack is run (e.g. by running `yarn build` or `yarn build:chrome`). We run a preprocessor on `manifest.json.src` to generate a manifest suitable to different browser targets.

I'll have a proper look tomorrow. Thank you for the very comprehensive issue!

**Comment by cameel** [comment]
Thanks! That explains the separate source package. Yeah, the automatically generated one is sufficient for building, I used it to build the xpi and it worked. I just thought that the other one was meant to be used instead.

In that case maybe you could just briefly document the purpose of the release artifacts somewhere (in the `README` maybe?). The source packages are now clear to me but I still have no idea what the .js files are for (e.g. `rikaichamp-options.js`). They're already present in the built .xpi so I guess I don't need them. Or do I? Are they alternate versions for some specific use case?

Alternatively, documenting the recommended way to build the xpi without cloning the whole repo (which package to use, how to avoid the husky issue, etc.) might be helpful to others who try that in the future.

**Comment by birtles** [comment]
> Thanks! That explains the separate source package. Yeah, the automatically generated one is sufficient for building, I used it to build the xpi and it worked. I just thought that the other one was meant to be used instead.

Great! In that case I will drop the separate source package altogether.

Here's what I've done:

* Make `utils/zip-src.js` include `tsconfig.json`
* Made CI no longer call it since we can rely on the GitHub generated source packages

> In that case maybe you could just briefly document the purpose of the release artifacts somewhere (in the `README` maybe?). The source packages are now clear to me but I still have no idea what the .js files are for (e.g. `rikaichamp-options.js`). They're already present in the built .xpi so I guess I don't need them. Or do I? Are they alternate versions for some specific use case?

`rikaichamp-options.js` etc. are actually for the bug reporting framework. Rikaichamp sends a bug report to Bugsnag whenever there is an unexpected bug and Bugsnag needs the source code for each version to be available publicly so it can correlate stack traces to the source and we use the GitHub release URL for that.

Perhaps the most useful place to put that explanation would be alongside each release?

I haven't automated the generation of release notes yet, however, so I'll have to remember to add that manually each time for now.

> Alternatively, documenting the recommended way to build the xpi without cloning the whole repo (which package to use, how to avoid the husky issue, etc.) might be helpful to others who try that in the future.

Yes, good idea. I suppose that too would be something to document along with each release? Perhaps each release could include the notes on how to build (e.g. `RIKAICHAMP_RELEASE=1 yarn install && yarn build`) and then the `README` could refer to that?

**Comment by cameel** [comment]
> Perhaps the most useful place to put that explanation would be alongside each release?

Oh, so that's what it's about. Personally, I would just put it in the README. I.e. I'd add a very short section about installation, basically telling users:
- to try getting it from their browser's addon store first (possibly with a direct link to the page in each store)
- that the packages are also available as `rikaichamp-x.y.z-<browser>.zip` on the release page
- to ignore the `rikaichamp-<something>.js` files on the release page because they're only needed for technical reasons (Bugsnag).

Having to include it each time in release notes sounds like too much hassle to me :)

> Yes, good idea. I suppose that too would be something to document along with each release? Perhaps each release could include the notes on how to build (e.g. RIKAICHAMP_RELEASE=1 yarn install && yarn build) and then the README could refer to that?

Again, I'd just chuck it in the README. For me it's the first place where I would look for it. Unless the build instructions differ wildly between releases, I'd only expect to find a summary of changes on the release page.

**Comment by birtles** [comment]
Thanks, that's great. I've added those suggestions to the README in 30a7acad26a4f85311fa73f3b0cc386556bcc89d. Hopefully that fixes this issue.

**Comment by cameel** [comment]
Nice. I think that solves the issue.

The only suggestion I would have is that maybe `yarn package` would be more appropriate than `yarn build` in the instructions. After all that gives you a plugin ready to install.

**Comment by birtles** [comment]
> The only suggestion I would have is that maybe `yarn package` would be more appropriate than `yarn build` in the instructions. After all that gives you an addon ready to install.

Good point! Fixed in 7832b369d06dcbc0125c26546a81570dc346a817
